# Getting the latitude/longitude of postalcodes

Importing needed packages

In [2]:
# import needed packages
import pandas as pd
from geopy.geocoders import Nominatim

Reading the dataframe *toPostalDf2.csv*, created by the later notebook

In [3]:
# reading the dataframe (from the last notebook)
toPostalDf = pd.read_csv('toPostalDf2.csv')
toPostalDf.head()

Unnamed: 0 Postcode      Borough                        Neighbourhood
0           0      M1B  Scarborough                       Rouge, Malvern
1           1      M1C  Scarborough  HighlandCreek, RougeHill, PortUnion
2           2      M1E  Scarborough     Guildwood, Morningside, WestHill
3           3      M1G  Scarborough                               Woburn
4           4      M1H  Scarborough                            Cedarbrae

In [4]:
toPostalDf.shape

(103, 4)

For each *Postcode* in the DataFrame, get the latitude and longitude and assign it to latitude/longitude list  
  
<div style="margin:10px; border:1px solid black">
    <div style="background-color:red; padding:10px"><strong>IMPORTANT</strong></div>
    <p style="padding:5px">I'm unable to fetch geopy, geocoder. It returns error every time after fetch the location of 2 postcode. So, I needed to use the CSV of the task assignment to get latitude and longitude of postalcodes.</p>
</div>

In [11]:
#geolocator = Nominatim(user_agent='me_agent')
#latitude = []
#longitude = []
# for every postcode, get latitude and longitude
#for postcode in toPostalDf['Postcode']:
#    address = '{}, Toronto, Ontario'.format(postcode)
#    print('Getting location of ', postcode)
#    location = geolocator.geocode(address)
    
    # get the location (loop fix for none error, from class)
#    while (location is None):
#        location = geolocator.geocode(address)
        
#    latitude.append(location.latitude)
#    longitude.append(location.longitude)

#print a sample of locations
#print(latitude[0:5])
#print(longitude[0:5])

Getting location of  M1B
Getting location of  M1C
Getting location of  M1E


GeocoderUnavailable: Service not available

Check if the lenght of latitude and longitude lists are the same of the Data Frame (ie 103 entries)

In [24]:
# check integrity (need to be the same length of shape of DF)
#print(len(latitude))
#print(len(longitude))

2
2


Assign the latitude and longitude to the DataFrame

In [15]:
# assign latitude and longitude to dataframe
#toPostalDf['Latitude'] = latitude
#toPostalDf['longitude'] = longitude
#toPostalDf.head()

Unnamed: 0 Postcode      Borough                        Neighbourhood  \
0           0      M1B  Scarborough                       Rouge, Malvern   
1           1      M1C  Scarborough  HighlandCreek, RougeHill, PortUnion   
2           2      M1E  Scarborough     Guildwood, Morningside, WestHill   
3           3      M1G  Scarborough                               Woburn   
4           4      M1H  Scarborough                            Cedarbrae   

    Latitude  longitude  
0  43.653963 -79.387207  
1  43.653963 -79.387207  
2  43.653963 -79.387207  
3  43.653963 -79.387207  
4  43.653963 -79.387207

<div style="background-color:red; padding:10px"><h3>Using the CSV of project assignment to get latitude and latitude</h3></div>

In [13]:
df = pd.read_csv('http://cocl.us/Geospatial_data')
df.columns = ['Postcode', 'Latitude', 'Longitude']
df.head()

Postcode   Latitude  Longitude
0      M1B  43.806686 -79.194353
1      M1C  43.784535 -79.160497
2      M1E  43.763573 -79.188711
3      M1G  43.770992 -79.216917
4      M1H  43.773136 -79.239476

<div style="background-color:red; padding:10px">Merging the datasets</div>

In [22]:
# merging the two dataframes
toPostalDf = toPostalDf.drop('Unnamed: 0', axis=1).set_index('Postcode').join(other=df.set_index('Postcode'))

In [23]:
toPostalDf.head()

Borough                        Neighbourhood   Latitude  \
Postcode                                                                
M1B       Scarborough                       Rouge, Malvern  43.806686   
M1C       Scarborough  HighlandCreek, RougeHill, PortUnion  43.784535   
M1E       Scarborough     Guildwood, Morningside, WestHill  43.763573   
M1G       Scarborough                               Woburn  43.770992   
M1H       Scarborough                            Cedarbrae  43.773136   

          Longitude  
Postcode             
M1B      -79.194353  
M1C      -79.160497  
M1E      -79.188711  
M1G      -79.216917  
M1H      -79.239476

Clean and save the dataframe

In [27]:
# cleaning the dataframe and saving it
toPostalDf.to_csv('toPostalDf2_ll.csv')
toPostalDf.head()

Borough                        Neighbourhood   Latitude  \
Postcode                                                                
M1B       Scarborough                       Rouge, Malvern  43.806686   
M1C       Scarborough  HighlandCreek, RougeHill, PortUnion  43.784535   
M1E       Scarborough     Guildwood, Morningside, WestHill  43.763573   
M1G       Scarborough                               Woburn  43.770992   
M1H       Scarborough                            Cedarbrae  43.773136   

          Longitude  
Postcode             
M1B      -79.194353  
M1C      -79.160497  
M1E      -79.188711  
M1G      -79.216917  
M1H      -79.239476

## END